### PR TITLE
Feature/events feed

### PR DESCRIPTION
**Pull request by HyPn0TiiK**

### CODE REVIEW

File: app/src/androidTest/java/com/example/bootcamp/MainActivityTest.java [hunk 1]
```diff
@@ -44,6 +50,20 @@ public void first_test(){
...
+        Intents.release();
+    }
+
+    @Test
+    public void feedActivityButtonWorks(){
+       /* Espresso.onView(ViewMatchers.withId(R.id.feed_view_button)).perform(click());
+        Espresso.onView(ViewMatchers.withId(R.id.feed_activity)).check(matches(isDisplayed()));*/
+        Intents.init();
+
+        String test_string = "This is a test";
```
Comment (khcr) [inline app/src/androidTest/java/com/example/bootcamp/MainActivityTest.java:62]: I think this line is unnecessary

File: app/src/test/java/PublicOrganizerTests.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import ncnf.organizer.PublicOrganizer;
+
+import static org.junit.Assert.*;
+
+public class PublicOrganizerTests {
+
+    private PublicOrganizer po = new PublicOrganizer("publicOrganizerTest");
+
+    @Test
+    public void getPublicOrganizerNameWorks(){assertEquals(po.getName(), "publicOrganizerTest");}
```
Comment (khcr) [inline app/src/test/java/PublicOrganizerTests.java:12]: Could use a constant string to store the name.

File: app/src/test/java/PublicEventTests.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import ncnf.event.EventType;
+import ncnf.event.PublicEvent;
+import ncnf.organizer.PublicOrganizer;
+
+import static org.junit.Assert.*;
+
+public class PublicEventTests {
+
+    private PublicOrganizer po = new PublicOrganizer("publicOrganizer");
+    private PublicEvent pe = new PublicEvent(po, EventType.Museum, "museumEvent", "EventData");
```
Comment (khcr) [inline app/src/test/java/PublicEventTests.java:14]: Could store the information in constants.

File: app/src/main/java/ncnf/main/MainActivity.java [hunk 4]
```diff
@@ -1,20 +1,39 @@
...
+
+        Button launchFeed = (Button) findViewById(R.id.feed_view_button);
+
+        launchFeed.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                feedView();
+            }
+        });
```
Comment (khcr) [inline app/src/main/java/ncnf/main/MainActivity.java:36]: Is it not cleaner/simpler to to add the click event directly in activity view ? (android:onClick="goToFeed")

File: app/src/main/java/ncnf/feed/FeedActivity.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    private RecyclerView.LayoutManager lManager;
+    private EventAdapter adapter;
+    private List<Event> eventList;
+    private static final String CHANNEL_NAME = "events_to_be_shown";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        //setContentView(R.layout.activity_feed);
+        setContentView(R.layout.activity_feed);
```
Comment (khcr) [inline app/src/main/java/ncnf/feed/FeedActivity.java:30]: We should maybe nest the component's views in a folder ? (e.g. layout/feed/main.xml)

File: app/src/androidTest/java/FeedActivityTests.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Rule
+    public ActivityScenarioRule<FeedActivity> testRule =
+            new ActivityScenarioRule<>(FeedActivity.class);
+
+    @Test
+    public void eventActivityOpens(){
+        Intents.init();
+
+        String test_string = "This is a test";
```
Comment (khcr) [inline app/src/androidTest/java/FeedActivityTests.java:43]: I think this line is unnecessary.

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -1,27 +1,54 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
-    package="com.example.bootcamp">
+    package="com.ncnf" >
+
+    <uses-permission android:name="android.permission.WAKE_LOCK" />
+
+    <uses-feature android:name="android.hardware.type.watch" />
```
Comment (khcr) [inline app/src/main/AndroidManifest.xml:8]: Why these permissions ?

File: app/build.gradle [hunk 8]
```diff
@@ -37,12 +37,20 @@ dependencies {
     implementation 'androidx.appcompat:appcompat:1.2.0'
     implementation 'com.google.android.material:material:1.3.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
+    implementation 'androidx.wear:wear:1.0.0'
```
Comment (khcr) [inline app/build.gradle:40]: Why a package that is for wearable device ?

**Review by khcr — APPROVED** [review]
Nice feature.
Documenting the code could be useful !

**Review by Arthelh — APPROVED** [review]
Reviewed changes made by @HyPn0TiiK. Everything looks good and code looks clean. Nothing needs major change in my opinion

**Pull request merged**